### ISSUE TITLE
Add ability to derive transparent newtypes

### ISSUE THREAD

**Issue by bdrobinson** [async-graphql/async-graphql#388]
## Description of the feature
Perhaps this feature already exists and I just missed it. But basically I'd like to be able to write code like this:

```rust
#[derive(async_graphql::SimpleObject)]
struct Product {
  weight: Weight,
}
struct Weight(f64);
```

and have it generate the following graphql schema:
```graphql
type Product {
  weight: Float!
}
```
ie, for `async_graphql` to treat the `Weight` newtype transparently. Is there a way I can do this? I'd imagine the API would be something like
```rust
#[derive(async_graphql::NewType)]
struct Weight(f64);
```

Thanks very much!

**Comment by sunli829** [comment]
Released in `v2.4.11`.

It would be great if you could help me add the description of `NewType` in the `Custom scalar` chapter in `Book`. 😁

**Comment by bdrobinson** [comment]
Incredible, thanks very much! Will have a go shortly 👍
